core/src/main/java/org/apache/calcite/sql/SqlBitwiseBinaryOperator.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.sql.fun.SqlLibraryOperators;
+import org.apache.calcite.sql.fun.SqlStdOperatorTable;
+import org.apache.calcite.sql.type.ReturnTypes;
+import org.apache.calcite.sql.type.SqlOperandTypeChecker;
+import org.apache.calcite.sql.validate.SqlValidator;
+import org.apache.calcite.sql.validate.SqlValidatorScope;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * <code>SqlBitwiseBinaryOperator</code> is a binary bitwise operator.
+ */
+public class SqlBitwiseBinaryOperator extends SqlBinaryOperator {
+
+  public SqlBitwiseBinaryOperator(
+      String name,
+      SqlKind kind,
+      int prec,
+      @Nullable SqlOperandTypeChecker operandTypeChecker) {
+    super(
+        name,
+        kind,
+        prec,
+        true,
+        ReturnTypes.BIGINT_NULLABLE,
```

Comment:
why bigint?
I think it should be the type of the operands - which should be coerced to have the same type.
Please note that some dialects such as MySQL allow some of these operations on BINARY types as well.
It's fine to start with just integers for now, but the solution should extend to BINARY eventually.

Reply:
BTW: Calcite already has aggregation functions BIT_AND and so on, and these return the same type as the input type. So the typing for these operators should be consistent with the aggregation functions.